### PR TITLE
Issue #11214: Specified violation message in JavadocParagraphCheck

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
JavadocParagraphCheck fix from Issue #11214

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
@Vyom-Yadav  Please share some feedback!

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java [hunk 1]
```diff
@@ -64,29 +69,35 @@
     InnerInputJavadocParagraphIncorrect anon = new InnerInputJavadocParagraphIncorrect() {
 
             /**
-         * <p>Some Javadoc. // violation
+         * <p>Some Javadoc. // violation 'Javadoc content should start from the next line after
+         * /\*\*.'
          *
-         * Some Javadoc. // violation above
+         * Some Javadoc. // violation above 'Javadoc content should start from the next line after
+         * /\*\*.'
```
Comment (Vyom-Yadav) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java:76]: It is difficult to read the message after manually escaping every character, please use `\Q` and `\E`.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java [hunk 2]
```diff
@@ -64,29 +69,35 @@
...
 
         /**
-         * Double newline. // violation below
+         * Double newline. // violation below 'Javadoc content should start from the next line
+         * after /\*\*.'
          *
          *
-         * Some Javadoc. //DOUBLE WARN AT TWO PREVIOUS LINES // violation above
+         * Some Javadoc. //DOUBLE WARN AT TWO PREVIOUS LINES // violation above 'Javadoc content
+         * should start from the next line after /\*\*.'
```
Comment (Vyom-Yadav) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java:100]: `// violation` should be the first comment for the parser to recognize it as a violation, having it as a second comment won't work.

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
Please fix CI issues, apply these changes for all similar items.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java [hunk 3]
```diff
@@ -9,52 +9,62 @@
 package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparagraph;
 
 /**
- * Some Javadoc. // 2 violations below
+ * Some Javadoc. // 2 violations below 'Javadoc content should start from the next line after
+ * /\*\*.'
```
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java:13]: Changes Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java]: Changes Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: > Also please don't wrap violation messages

Not quite done.

**Also you are using wrong violation messages**, you are using violation messages of another check (JavadocContentLocationCheck), violation messages for JavadocParagraph are [these](https://checkstyle.sourceforge.io/config_javadoc.html#JavadocParagraph_Violation_Messages)

Please change it to use correct violation messages and address other items in a similar manner after doing these changes.

**Comment by Vyom-Yadav** [comment]
Also please remove fixes from your PR description, the issue has a lot to do before being completely fixed.

**Comment by Vyom-Yadav** [comment]
Also please remove fixes from your PR description, the issue has a lot to do before being completely fixed.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java [hunk 4]
```diff
@@ -9,52 +9,66 @@
 package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparagraph;
 
 /**
- * Some Javadoc. // 2 violations below
+ * Some Javadoc. // violation '\Q Javadoc content should start from the next line
+ * after /\*\*.\E'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java:13]: Why you add a extra content in input file?

If 2 violation on same line happened we do not validate message, as we not know where which to choose.

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: Violation message Changed

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 4 above)
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: You mean,  when we have `// 2 violations` or `// 2 violation below` or `// 2 violation above` then we should not validate a message, right?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java]: Only use `\Q \E` where you need to escape many regex characters, for example if some violation message is like-
```
variable name must match the pattern ^[a-zA-Z_$][a-zA-Z_$0-9]*$
```
you can see at the end there are lot of regex characters, escaping them manually would make the violation message difficult to read and would also create a lot of noise in the violation message, so to overcome this problem you can simply change it to-
```
variable name must match the pattern \Q ^[a-zA-Z_$][a-zA-Z_$0-9]*$ \E
```
Don't use `\Q \E` when there are no special regex characters, for eg-
```
tag should be placed immediately before the first word, with no space after
```
There are no characters that need to be escaped so usage of `\Q \E` is unnecessary, please make these changes in all relevant cases.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: > Also please don't wrap violation messages.

Please don't wrap violation messages, keep them on a single line.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 4 above)
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: Yes, you can go through https://github.com/checkstyle/checkstyle/issues/11214#issue-1106705357 once again, it mentions that, also you can go through other similar PR's, they will help you in making changes.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect2.java]: Thanks a lot @Vyom-Yadav for the easy explaination.

**Comment by Rahulkhinchi03** [comment]
GitHub, rebase

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 4 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: Can we do:
```
// Violation 2 lines below, `message 1`
// Violation below, `message 2`
```
?

**Comment by romani** [comment]
Before pushing to GitHub please make sure that `mvn verify ` passing on your local

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 4 above)
Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: @romani I believe we can't do that as per the changes suggested by @Vyom-Yadav at https://github.com/checkstyle/checkstyle/issues/11214#issue-1106705357

**Comment by romani** [comment]
@Rahulkhinchi03 , why you close PR ?
we can merge your changes to make step forward, even it is not ideal.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java (hunk 4 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: did you try ?
I do not see issue that claim that it is not possible.

Comment (Rahulkhinchi03) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparagraph/InputJavadocParagraphIncorrect.java]: I tried to do it and it happened. You were right.

**Comment by Rahulkhinchi03** [comment]
> @Rahulkhinchi03 , why you close PR ?
> we can merge your changes to make step forward, even it is not ideal.

I believe I created a mess. I am creating a new PR with fresh changes.

**Pull request closed without merging**